kits/rosie/rosie_demo.cpp

```diff
@@ -0,0 +1,513 @@
+  //    % Get the initial feedback objects that we'll reuse later
+  GroupFeedback latest_phone_fbk(phone_group->size());// = phone_fbk;
+  phone_group->getNextFeedback(latest_phone_fbk);
+  
+  //    %%
+  //    %%%%%%%%%%%%%%%%%%%%%%%
+  //    % Begin the demo loop %
+  //    %%%%%%%%%%%%%%%%%%%%%%%
+  //    
+  //    % Start background logging
+  if(enable_logging){
+    robot_group->startLog("logs");
+    phone_group->startLog("logs");
+  }	
+  //    % This outer loop is what we fall back to anytime we 're-home' the arm
+  while(true){ 
+    GroupFeedback fbk(robot_group->size());
+    GroupCommand cmd(robot_group->size());
+    robot_group->getNextFeedback(fbk);
+
+    //        % Exaggerate Z-Axis by 2x, X-Y are 1-to-1.
+    Eigen::Vector3d xyz_scale;
+    xyz_scale << -1, -1, -2; 
+    //
+    //        % Move to current coordinates
+    Eigen::Vector3d xyz_target_init;
+    xyz_target_init << 0.3, 0.0, 0.3;
+    //        % Gripper down
+    Eigen::MatrixXd rot_mat_target_init(3,3);
+    rot_mat_target_init << -1,0,0, 0,1,0, 0,0,-1; 
+
+    Eigen::VectorXd arm_fbk_pos(arm_dofs.size());
+    for(size_t i = 0; i < arm_fbk_pos.size(); ++i){
+      arm_fbk_pos[i] = fbk.getPosition()[arm_dofs[i]];
+    }
+    Eigen::VectorXd ik_result_joint_angles(arm_dofs.size());
+  
+    arm_model->solveIK(arm_ik_seed_pos, ik_result_joint_angles, robot_model::EndEffectorPositionObjective(xyz_target_init), robot_model::EndEffectorSO3Objective(rot_mat_target_init));
+    const size_t len = arm_dofs.size();  
+    Eigen::MatrixXd position(len,2);
+    position.col(0) = arm_fbk_pos;
+    position.col(1) = ik_result_joint_angles;
+    Eigen::VectorXd time(2);
+    time << 0, arm_min_traj_duration;
+  
+    auto arm_traj = hebi::trajectory::Trajectory::createUnconstrainedQp(time, position);
+
+    auto t0 = fbk.getTime();
+    auto t = 0.0; 
+  
+    while(t < arm_min_traj_duration){
+      //
+      if(!robot_group->getNextFeedback(fbk)) printf("no feedback recieved\n");
+      auto temp_t = fbk.getTime();
+      t = (fbk.getTime() - t0) > arm_min_traj_duration ? arm_min_traj_duration : fbk.getTime() - t0;
+
+      getTrajectoryCommand(arm_traj, cmd, arm_dofs, t);
+  
+      //         TODO:   dynamicsComp
+      Eigen::VectorXd masses;
+      arm_model->getMasses(masses);
+      Eigen::VectorXd effort(arm_dofs.size());
+      auto base_accel = fbk[base_num_wheels].imu().accelerometer().get();
+      Eigen::Vector3d gravity(-base_accel.getX(),
+                     -base_accel.getY(),
+                     -base_accel.getZ());
+      Eigen::VectorXd arm_positions(arm_dofs.size());
+      for(size_t i = 0; i < arm_positions.size(); ++i){
+        arm_positions[i] = fbk.getPosition()[arm_dofs[i]];
+      }
+      effort = hebi::util::getGravityCompensationEfforts(*arm_model, masses, arm_positions, gravity);
+      /*TODO add dynamic_comp*/
+      setEffortSegment(cmd, arm_dofs, effort + arm_effort_offset);
+      robot_group->sendCommand(cmd);
+    }
```

Comment:
Note for future refactor -- we can probably encapsulate some of these lines in this loop to reuse.